Userland/Utilities/file.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2021, Valtteri Koskivuori <vkoskiv@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/MappedFile.h>
+#include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/FileStream.h>
+#include <LibCore/MimeData.h>
+#include <LibGfx/ImageDecoder.h>
+#include <stdio.h>
+#include <unistd.h>
+
+static Optional<String> placeholder(String description, String& path)
+{
+    (void)path;
+    return description;
+}
+
+//FIXME: Ideally Gfx::ImageDecoder could tell us the image type directly.
+static Optional<String> image_details(String description, String& path)
+{
+    auto file_or_error = MappedFile::map(path);
+    if (file_or_error.is_error()) {
+        return {};
+    }
+
+    auto& mapped_file = *file_or_error.value();
+    auto image_decoder = Gfx::ImageDecoder::create((const u8*)mapped_file.data(), mapped_file.size());
+
+    if (!image_decoder->is_valid())
+        return {};
+
+    StringBuilder builder;
+    builder.append(description);
+    builder.append(String::formatted(", {} x {}", image_decoder->width(), image_decoder->height()));
+    return builder.to_string();
```

Comment:
Don't think we need a string builder for this:

```suggestion
    return String::formatted("{}, {} x {}", description, image_decoder->width(), image_decoder->height());
```